shell/services/node/node_service.cc

```diff
@@ -89,64 +80,19 @@ void NodeService::Initialize(node::mojom::NodeServiceParamsPtr params) {
   // Add entry script to process object.
   gin_helper::Dictionary process(env->isolate(), env->process_object());
   process.SetHidden("_serviceStartupScript", params->script);
-  process.SetHidden("_postMessage",
-                    base::BindRepeating(&NodeService::PostMessage,
-                                        weak_factory_.GetWeakPtr()));
 
-  // Load everything.
-  node_bindings_->LoadEnvironment(env);
+  // Setup microtask runner.
+  js_env_->OnMessageLoopCreated();
 
   // Wrap the uv loop with global env.
   node_bindings_->set_uv_env(env);
 
-  // Setup microtask runner.
-  js_env_->OnMessageLoopCreated();
+  // Load everything.
+  node_bindings_->LoadEnvironment(env);
```

Comment:
!, why do we need to switch the order of these two?

Reply:
I should add a comment for this change, the previous order led to a crash when I tried to implement https://github.com/electron/electron/blob/014d6e8b897fedba1222a36de0227e40b2d606eb/shell/services/node/node_service.cc#L65-L73 using a sample script that threw an uncaught exception on start.

```
// test.js
nonExistentFunc();
``` 

The script actually gets loaded during `LoadEnvironment` and when the error happens, it will attempt to abort the process which is then caught by our custom exit handler but at this time we have not setup the microtask runner https://github.com/electron/electron/blob/014d6e8b897fedba1222a36de0227e40b2d606eb/shell/browser/javascript_environment.cc#L292, so `js_env_->OnMessageLoopDestroying()` will crash. We have to setup the runner before loading the Node.js environment.

This is not an issue for the main process because we don't try to abort the process for uncaught exceptions.

Reply:
Refs https://github.com/electron/electron/pull/34980/commits/a82d3223ede85df14b97780ef2bc1cce21e3aa39